src/ui/application-window.vala

```diff
@@ -268,6 +294,34 @@ namespace Peek.Ui {
         }
       }
     }
+    
+    
+    
+
+    [GtkCallback]
+    private void on_gif_button_clicked (Button source) {
+    	recorder.output_format="gif";
+	    record_button.set_label("Record as GIF");
+    	pop_format.hide();
+    	}
+    [GtkCallback]
+    private void on_apng_button_clicked (Button source) {
+    	recorder.output_format="apng";
+    	record_button.set_label("Record as APNG");
+    	pop_format.hide();
+    	}
+    [GtkCallback]
```

Comment:
As this functionality is repeating for all buttons (and maybe there will be even more in the future) I would wrap it in a select_format () helper function, something like this:

```
private string get_format_name (format) {
    switch (format) {
      case OUTPUT_FORMAT_APNG: return _("APNG");
      case OUTPUT_FORMAT_GIF: return _("GIF");
      case OUTPUT_FORMAT_MP4: return _("MP4");
      case OUTPUT_FORMAT_WEBM: return _("WebM");
      default: return "";
    }
}

private void select_format (format) {
    recorder.config.output_format = format;
    var format_name = get_format_name (format);
    record_button.set_label (_("Record as %s").printf (format_name));
    pop_format.hide ();
}
```

That's easier to extend and change in the future.